components/mpas-ocean/src/mode_forward/mpas_ocn_time_integration_split.F

```diff
@@ -798,7 +796,7 @@ subroutine ocn_time_integrator_split(domain, dt)!{{{
             !$omp parallel
             !$omp do schedule(runtime) &
             !$omp private(cell1, cell2, k, thicknessSum)
-            do iEdge = nEdgesOwned+1, nEdgesArray(4)
+            do iEdge = 1, nEdgesArray(size(nEdgesArray)-1)
```

Comment:
Any reason we're not using nEdgesAll here? Would save a size evaluation. Also nEdgesArray has been deprecated in ocn_mesh and does not exist on the device, so if a halo width is needed, please use the nEdgesHalo array instead.

Reply:
`nEdgesAll` includes the outer edge of the last halo layer, which we don't need and caused an error in the loop. I'll try it with `nEdgesHalo` - thanks for pointing that out.

Reply:
I changed to `nEdgesHalo(config_num_halos+1)` and verified that the value is 4.